third_party/move/move-compiler-v2/tests/checking/typing/block_empty_invalid.exp

```diff
@@ -1,18 +1,18 @@
 
 Diagnostics:
-error: expected `u64` but found `()` in unit expression
+error: expected `u64` but found `()` in expression
```

Comment:
I'm concerned about the potential confusion between {} and ().  Are these the same value?  Structs are not the same as Tuples.  Oh, this is an empty expression block {} returning ().

In any case, better to say something like:

**error: expression of unit type '()' not compatible with type u64.**

Even the old error messages weren't great.  You don't expect 'u64', you expect an expression of type 'u64' here.

Reply:
How about "Expected `uint64` but found expression of type `()`."

Or maybe "value of type `()`"

Reply:
Again, this not a manually created message which you can just change like you are suggesting:

- The message the type unifier generates is `expected X but found Y`. The unifier also produces a couple of different messages, this is just one of them.
- Then this message as generated by the unifier (whatever it is) is augmented by a suffix 'in XYZ' which gives context in which the type unification failure occured.

So its `generic_type_unification_error + context_message`. 

Also please note that little of this is new but just the type checker of the specification language how it has been working since a while, with little complaints. So I did not add all this in this PR, but only adapt it.

Reply:
After reviewing the error messages, @brmataptos , I actually removed all those `in XYZ context` message suffices. The reason is that the last fix based on your feedback -- where the `^^` was not complete or on the wrong place -- actually was a bit more fundamental. Basically, when we add a variable binding with a constraint (say `x where SomeReference(y)`), and this constraint is later resolved as unsat during type unification, we need to report the error back at the point where the constraint was created -- not where we perform the unification step. This makes all those 'in lvalue', 'in constant', `in pack expression' messages potentially inaccurate. I decided to better drop them for now.

Reply:
Ok, I see it now.